File: PSP/CajaFuerte/cliente/src/main/java/ui/MainFX.java
package ui;

import jakarta.enterprise.event.Observes;
import jakarta.inject.Inject;
import javafx.fxml.FXMLLoader;
import javafx.scene.Parent;
import javafx.scene.Scene;
import javafx.stage.Stage;
import ui.common.ConstantesUI;
import ui.pantallas.pantallamain.PantallaMainController;

import java.io.IOException;
import java.util.ResourceBundle;

public class MainFX {
    @Inject
    FXMLLoader fxmlLoader;

    public void start(@Observes @StartupScene Stage stage) {
        try {
            ResourceBundle r = ResourceBundle.getBundle(ConstantesUI.BUNDLE);
            fxmlLoader.setResources(r);
            Parent fxmlParent = fxmlLoader.load(getClass().getResourceAsStream(ConstantesUI.FXML_PANTALLA_MAIN_FXML));
            PantallaMainController controller = fxmlLoader.getController();
            controller.setStage(stage);

            stage.setScene(new Scene(fxmlParent));
            //stage.getScene().getStylesheets().add(getClass().getResource("/css/style.css").toExternalForm());
            stage.setTitle(r.getString(ConstantesUI.TITULO));
            stage.show();
        } catch (IOException e) {
            e.printStackTrace();
            System.exit(0);
        }
    }
}

File: PSP/arreglitos/src/main/kotlin/ui/MainFX.kt
package ui

import javafx.application.Application
import javafx.fxml.FXMLLoader
import javafx.scene.Scene
import javafx.scene.layout.BorderPane
import javafx.stage.Stage
import ui.pantallas.mainscreen.MainController

class MainFX : Application() {

    override fun start(primaryStage: Stage?) {
        try {
            val loaderMenu = FXMLLoader(javaClass.getResource("/fxml/PantallaMain.fxml"))
            val root: BorderPane? = loaderMenu.load()
            val controller: MainController = loaderMenu.getController()
            controller.setStage(primaryStage)
            val scene = Scene(root)
            primaryStage!!.title = "ClienteGraphQL"
            primaryStage.scene = scene
            primaryStage.isResizable = true
            primaryStage.minWidth = 600.0
            primaryStage.minHeight = 800.0
            primaryStage.isMaximized = true
            primaryStage.show()
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}

fun main(args: Array<String>) {
    Application.launch(MainFX::class.java, *args)
}
